### PR TITLE
write last rows in PrettyMonoBlock

### PR DESCRIPTION
**Pull request by Volodyachan**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Outputs the last 10 rows if `output_format_pretty_max_rows` is exceeded
If `output_format_pretty_max_rows < 10`, outputs `output_format_pretty_max_rows / 2` last rows

Example:
```
SELECT *
FROM system.numbers
LIMIT 10000
FORMAT PrettyMonoBlock
SETTINGS output_format_pretty_max_rows = 9

Query id: 958bc7a4-d926-4946-a56a-dcc94f699e06

       ┏━━━━━━━━┓
       ┃ number ┃
       ┡━━━━━━━━┩
    1. │      0 │
       ├────────┤
    2. │      1 │
       ├────────┤
    3. │      2 │
       ├────────┤
    4. │      3 │
       ├────────┤
    5. │      4 │

       ⋮

 9997. │   9996 │
       ├────────┤
 9998. │   9997 │
       ├────────┤
 9999. │   9998 │
       ├────────┤
10000. │   9999 │
       └────────┘
  Showed 9.
```

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 5b6bb60658d3a7337d4d55f4748be71504e9bcab with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64200/5b6bb60658d3a7337d4d55f4748be71504e9bcab/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by jkartseva — CHANGES_REQUESTED** [review]

File: src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp [hunk 1]
```diff
@@ -482,6 +501,12 @@ void PrettyBlockOutputFormat::writeMonoChunkIfNeeded()
 {
     if (mono_chunk)
     {
+        if (mono_chunk.getNumRows() > format_settings.pretty.max_rows)
+        {
+            if (mono_chunk.getNumRows() > 10)
```
Comment (jkartseva) [inline src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp:506]: Please name the constant, e.g. `constexpr size_t last_rows_num = 10`
Alternatively, `auto last_rows_num = std::min(format_settings.pretty.max_rows / 2, 10)`

File: src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp [hunk 2]
```diff
@@ -493,7 +518,10 @@ void PrettyBlockOutputFormat::writeSuffix()
 
     if (total_rows >= format_settings.pretty.max_rows)
```
Comment (jkartseva) [inline src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp:519]: When `output_format_pretty_max_rows` equal to `LIMIT` it writes "Showed first":
```
SELECT *
FROM system.numbers
LIMIT 3
FORMAT PrettyMonoBlock
SETTINGS output_format_pretty_max_rows = 3

Query id: 7f513247-acf0-4f01-95d0-de705b1a90ea

   ┏━━━━━━━━┓
   ┃ number ┃
   ┡━━━━━━━━┩
1. │      0 │
   ├────────┤
2. │      1 │
   ├────────┤
3. │      2 │
   └────────┘
  Showed first 3.

3 rows in set. Elapsed: 0.002 sec.
```

It seems the condition should be `total_rows > format_settings.pretty.max_rows`

**Comment by jkartseva** [comment]
@Volodyachan please take a look at the review comments. Build failure ought to be fixed: https://github.com/ClickHouse/ClickHouse/actions/runs/9224891757/job/25387973912#step:8:11008

**Comment by alexey-milovidov** [comment]
I thought about implementing it in a simpler way. I will submit another PR.

**Pull request closed without merging**